graphql_compiler/tests/test_compiler.py

```diff
@@ -4191,6 +4224,64 @@ def test_fold_after_traverse(self):
         check_test_data(self, test_data, expected_match, expected_gremlin, expected_sql,
                         expected_cypher)
 
+    def test_fold_after_traverse_different_types(self):
```

Comment:
what is different types referring to here? Location vs. Animal? Should other languages use this test?

Reply:
Yes, this test was to make debugging easier b/c you can see when you cross the edge that you've also changed types. I don't know if its necessary for any of my tests to be written into the other languages but if possible I'd like to put that into a different PR b/c I don't really understand Gremlin at all.